pages/glossary/text-same-language.md

```diff
@@ -0,0 +1,33 @@
+---
+title: Text with the same Programmatic Language
```

Comment:
I'd rather keep the definition title as concepts rather than technology specific stuff (even if the definition is technology specific).

Among other, the definition is not restricted to elements that have a `lang` attribute in the first place. In `<div>foo</div>`, "foo" is text with the same programmatic language as the `div`, even though there is no `lang` attribute implied.

Reply:
Right, but logically in the following example "foo" and "bar" are texts with the same programmatic language, but your definition says they wouldn't be. I don't think the name reflects what the definition actually does.

```html
<i lang="nl">foo</i>
<b lang="nl">bar</b>
```

Reply:
Fair point. I was seeing "same" as physical rather than structural equality, but it doesn't make sense in common language…

Reply:
After more thoughts and work into this, I don't think the definition should be tied to a `lang` attribute as it ripples down further definitions.

This definition is essentially extracting the list from https://act-rules.github.io/rules/ucwvc8#default-page-language and is intended to define the default language of elements. If this definition is tied to a `lang` attribute, it means we only define "default language" of elements with a `lang` attribute which make the definition of "default language" extremely un-natural.

I'm struggling to find a better name for the definition, and I'm definitely welcoming ideas, but I'm very reluctant to tie it to the presence of a `lang` attribute.

Reply:
But the definition ties it to the presence of a `lang` attribute, doesn't it?
I've been trying to come up with a proposal for a name, but I can't seem to improve on @WilcoFiers suggestion. Eventually "Text affected by the same `lang` attribute"?

Reply:
> But the definition ties it to the presence of a `lang` attribute, doesn't it?

No, not really (only as a descendants limit).

In 
```html
<div>
Some text in English.
<span lang="fr">Du texte français</span>
<span>More text in English.</span>
```
The first sentence, and the second `span` is "text with the same programmatic language" as the `div`. And that is true whatever the context in which the `div` is put.

If we move to "text affected by a `lang` attribute", then only the text in the first `span` will be affected by a `lang` attribute, but that changes with context (if the full snippet is inside a `<div lang="en">` then text becomes affected by that new `lang` attributes).

Now, the **rules** only use that on elements that do have a `lang` attribute, so it is not a big deal at that level.
**But** this is used in the definition of most common and default language. So it means these definitions would then only apply to elements that have a `lang` attribute. I don't like saying that in
```html
<div>
Some text in English.
<span lang="fr">Du texte français</span>
<span>More text in English.</span>
```
the `div` has no "default language" (just because it has no `lang` attribute). That makes the name "default language" counter-intuitive and we're just pushing the problem of poor definition name further…

Reply:
Weirder still, the below example does not have the same programmatic language (according to this definition).
```html
<div lang="en">Some text <span lang="en">in English</span>.<div>
```
Should we compare the value of the `lang` attributes when they exist to check if they have the same value? Then we could go all the way up to the `body` element which would address Wilco's example. And would make the name much more suitable.

Reply:
Yes, that's why "same programmatic language" is bad in the first place… 

I'm not too keen on looking at other element with the exact same `lang` attribute on the page because that is absolutely not what the rules need. In your example, when the rule checks whether `<div lang="en">` is correct, it really doesn't matter than there is a `<span lang="en">` inside, or another `<div lang="en">` on the page…

But maybe that leads to the way out 🎉 

What about "text/elements affected by the same `lang` attribute" (mixing both names).
That would work on stuff that has no `lang` attribute, and that would make it clear that "same" is about physical identity rather than structural one (`<div lang="en">` and `<div lang="en">`) may have the same programmatic language (English), but not the same `lang` attribute).

This should also work out that in
```html
<div>
Some text in English.
<span lang="fr">Du texte français</span>
<span>More text in English.</span>
```
the second `span` is "affected by the same `lang` attribute" than the `div`, even though there is none currently affecting them, whatever the context we put this where a `lang` attribute affects the `div`, that `lang` attribute would also affect the second `span` but not the first (even if it's also a `lang="fr"`, the first `span` doesn't care).

And if the definition can work on stuff without a `lang` attribute on them, then anything may have a default or most common language and that is cool for me 😎

Reply:
@Jym77 I have to admit I couldn't understand what you are proposing here :-(
Perhaps we can talk about this after the TF/CG meeting later today, if you have time?

Anyway, I've though about this a bit more and have another suggestion, looking at this from another perspective. What if instead of trying to identify text that has the same language, we focus on text that does not change the language? Something like "Inclusive text that does not change the programmatic language"?

Reply:
@carlosapaduarte I was (ultimately) suggesting to rename the def to "text/element affected by the same `lang` attribute". But I like your idea better 😄 ("inclusive text" is maybe a bit misleading…)

Reply:
@carlosapaduarte @WilcoFiers I end up with "text/element inheriting its programmatic language" from an element.

This is in the same idea of what Carlos was suggesting, but swapping the "does not change" by "receive from" which I feel is closer to the idea. I also felt that "inclusive text" was too close to "inclusive language" which has a totally different meaning (especially when talking about disabilities…)

I think that this is good. Especially, the "element inheriting the programmatic language" works well by using the same vocabulary as CSS inheritance since programmatic language is pretty much determined like an inherited CSS property… So the analogies that this could trigger will actually help understand the definition…